File: Composition/CS/Composition/Algorithm/Composition/SummingStrategy.cs
using System.Collections.Generic;
using System.Linq;

namespace Algorithm.Composition
{
    public class SummingStrategy : IAggregationStrategy
    {
        public Measurement Aggregate(IEnumerable<Measurement> measurements)
        {
            return new Measurement { X = measurements.Sum(m => m.X), Y = measurements.Sum(m => m.Y) };
        }
    }
}

File: Composition/Java/Composition/src/main/java/algorithm/composition/SummingStrategy.java
package algorithm.composition;

import static algorithm.MeasurementUtil.sumX;
import static algorithm.MeasurementUtil.sumY;

import java.util.Collection;

import algorithm.Measurement;

public class SummingStrategy implements IAggregationStrategy {

	@Override
	public Measurement aggregate(Collection<Measurement> measurements) {
		return new Measurement(sumX(measurements), sumY(measurements));
	}
}
